scripts/dist_build.sh

```diff
@@ -28,24 +28,26 @@ make get_vendor_deps
 # ldflags: -s Omit the symbol table and debug information.
 #	         -w Omit the DWARF symbol table.
 echo "==> Building..."
-"$(which gox)" \
```

Comment:
I still don't understand why we're mixing up the build process with the packaging process. Assume we want to build a new version:
1. We build using `make build`
2. Run tests
3. `Run `make dist` that rebuilds again?
4. Deploy the zips

The tested binaries should be packaged and we shouldn't need to build again. This way we're packaging untested binaries.

Reply:
1. Run tests
2. Run ./scripts/release.sh (dist.sh + publish.sh)

Reply:
MAybe I'm missing something, please clarify. This is how I see it: `Run tests` builds a binary and tests it. We throw it away. `release.sh` runs `dist.sh` which also build a binary which was untested. That gets packaged.
I think a binary should only be packaged if it ran through testing with success.

Reply:
> We throw it away. release.sh runs dist.sh which also build a binary which was untested

but the code stays the same!

Reply:
Well, we can't really prove that, can we? The tests threw away the binary without taking a note of the hash and although we want deterministic builds, currently we've only done a few tests on that. For example we wouldn't know if any environmental change happened in the test environment or the packaging environment.

Reply:
We can't _assume_ that the code stays the same, we have to verify, if we build multiple times. Building multiple times also lengthens the whole process for no apparent reason.

Reply:
agree. we can checksum the code and use it across steps

Reply:
The dist script produces cross-compiled scripts.  We can't test cross-compiled binaries anyways.
I'm not sure that checksuming the code helps either.
I don't see the problem with building and throwing it away.  Build is for developers.  Dist is for presumably the dist-bot.  The dist-bot need not run build first.

Reply:
> The tests threw away the binary without taking a note of the hash and although we want deterministic builds, currently we've only done a few tests on that.

Deterministic builds assume a deterministic environment.  `make build` doesn't ensure any deterministic environment, we ensure that by running `make build` in a given Docker container environment.  No need to take the hash of the build.